client/src/components/FileUpload.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import Dropzone from '../components/Dropzone';
+import Button from '../components/Button';
+import Warning from '../components/Warning';
+import H1 from '../components/H1';
+import H2 from '../components/H2';
+import Body from '../components/Body';
+
+const FileUpload = ({ next }: { next: () => void }) => {
+  const [files, setFiles] = useState<File[]>([]);
+  const [errors, setErrors] = useState<(string | null)[]>([]);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  const hasErrors = () => errors.filter((error) => error !== null).length > 0;
+
+  useEffect(() => {
+    const login = async () => {
+      const res = await fetch('http://localhost:4567/api/v1/login', {
+        method: 'POST',
+        credentials: 'include',
+      });
+      if (!res.ok) {
+        throw new Error('HTTP error ' + res.status);
+      }
+      setIsLoggedIn(true);
+    };
+    if (!isLoggedIn) {
+      login();
+    }
+  }, []);
+
+  const uploadFiles = () => {
+    if (!isLoggedIn) {
+      throw new Error('not authenticated');
+    }
+    const reader = new FileReader();
+
+    files.forEach(async (file) => {
+      reader.readAsArrayBuffer(file);
+      reader.onload = async () => {
+        const res = await fetch(
```

Comment:
should this (and the fetch above) be in a try/catch block to handle errors (besides checking status which you already did)?

Reply:
I think it should be fine because the only two possibilities are that the status is ok or that it has an error, so that if check should handle it

<img width="1002" alt="image" src="https://github.com/davidpak/SmartNote/assets/25410985/fb4ecff5-3c1c-4f5e-82bc-ca7edbde9196">